### PR TITLE
Feat/user balance

### PR DESCRIPTION
**Pull request by sorrelbri**
No styling applied yet: 
Screen Shot 2020-07-27 at 6 44 03 PM
Working through data flow for payment intent creation.

### CODE REVIEW

File: client/src/pages/Checkout.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  Stripe,
+  useStripe,
+  useElements,
+} from "@stripe/react-stripe-js";
+import { loadStripe } from "@stripe/stripe-js";
+import { createPaymentIntent } from "../services";
+
+const showCart = (cart) => {
+  // only show most recent
+  const item = cart[0];
```
Comment (skassam21) [inline client/src/pages/Checkout.js:15]: I guess we only have one type of cart item right now or I'd say just map through the items

File: client/src/pages/Checkout.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          <td className="">${item.unitCost * item.quantity}</td>
+        </tr>
+      </tbody>
+    </table>
+  );
+};
+
+const stripePromise = loadStripe("key");
+const Checkout = ({ state, dispatch }) => {
+  usePageLoaded(dispatch);
```
Comment (skassam21) [inline client/src/pages/Checkout.js:38]: I actually don't really see the purpose of this hook, is it really necessary at this time to have it? It's just to be there when the page is loaded?

File: client/src/pages/Balance.js [hunk 3]
```diff
@@ -16,16 +17,25 @@ import AddIcon from "@material-ui/icons/Add";
...
-    if (id === "incr") return setTopUp(topUp + 1);
+    if (id === "incr") return setTopUp(() => topUp + 1);
     if (id === "decr" && topUp > 0) return setTopUp(topUp - 1);
   };
 
-  const handleCheckout = (e) => {};
+  const handleCheckout = (e) => {
+    dispatch({
+      type: "ADD_ITEM_TO_CART",
+      item: { name: "Review Credits", quantity: topUp, unitCost: 10 },
```
Comment (skassam21) [inline client/src/pages/Balance.js:35]: Can we make details about this item as constants somewhere - i.e. the name, the unit cost and the initial value? Something in some sort of constants / utils file you can reference

File: client/src/reducers/index.js [hunk 4]
```diff
@@ -9,6 +9,19 @@ export default (state, action) => {
...
+    }
+    case "CREATE_PAYMENT_INTENT": {
+      return { ...state, loading: false };
+    }
+    case "CREATE_PAYMENT_INTENT_ERROR": {
+      return { ...state, loading: false };
+    }
+    case "FINISH_LOAD": {
+      return { ...state, loading: false };
+    }
```
Comment (Affeeq) [inline client/src/reducers/index.js:24]: It seems there would be similar cases in the reducer. eg CREATE_PAYMENT_INTENT and FINISH_LOAD. Would it be better to reuse reducers?

File: client/src/pages/Checkout.js (hunk 2 above)
Comment (sorrelbri) [reply client/src/pages/Checkout.js:38]: In this case I wanted to lock input once the user submits the cart on `/balance` and unlock once the Checkout component mounts. I assumed this kind of control might be useful in other areas of the app where data flows through multiple pages and it was a pretty trivial to put in a hook.

File: client/src/pages/Balance.js (hunk 3 above)
Comment (sorrelbri) [reply client/src/pages/Balance.js:35]: Totally, I'm assuming at a certain point there will be BE data to request for this.

File: client/src/reducers/index.js (hunk 4 above)
Comment (sorrelbri) [reply client/src/reducers/index.js:24]: The `CREATE_PAYMENT_INTENT` and `CREATE_PAYMENT_INTENT_ERROR` are stubs. I'm working on the BE endpoints on my next PR, so I'm not entirely sure what the data will look like yet.

File: client/src/pages/Checkout.js (hunk 1 above)
Comment (sorrelbri) [reply client/src/pages/Checkout.js:15]: That's the idea, I wanted to create something that would be quick, but easy to extend.

File: client/src/pages/Balance.js [hunk 5]
```diff
@@ -14,18 +15,28 @@ import {
...
 import AddIcon from "@material-ui/icons/Add";
 import RemoveIcon from "@material-ui/icons/Remove";
+import { reviewCredits } from "../utils/itemLookup";
 
 const BalancePage = ({ state, dispatch }) => {
+  const priorTopUp = state.cart.find((item) => item.name === "Review Credits")
+    ?.quantity;
   const user = useContext(userContext);
-  const [topUp, setTopUp] = useState(2);
+  const [topUp, setTopUp] = useState(() => priorTopUp || 2);
```
Comment (skassam21) [inline client/src/pages/Balance.js:24]: get 2 from itemLookup

File: client/src/pages/Balance.js [hunk 6]
```diff
@@ -14,18 +15,28 @@ import {
 } from "@material-ui/core";
 import AddIcon from "@material-ui/icons/Add";
 import RemoveIcon from "@material-ui/icons/Remove";
+import { reviewCredits } from "../utils/itemLookup";
 
 const BalancePage = ({ state, dispatch }) => {
+  const priorTopUp = state.cart.find((item) => item.name === "Review Credits")
```
Comment (skassam21) [inline client/src/pages/Balance.js:21]: get the name from itemLookup

**Pull request merged**